### PR TITLE
Trace system

### PR DESCRIPTION
**Pull request by olivierschirm**
To install lttng: https://lttng.org/docs/v2.10/#doc-installing-lttng

To compile using the trace system: add -t or --trace

To test tracing some instances :
lttng create "Nom de la session"
lttng enable-event --userspace Nit_Compiler:Object_Instance
lttng start
--> run your program
lttng stop

To read your CTF trace file:
babeltrace ~/lttng-traces/nom_de_la_session

### CODE REVIEW

File: clib/traces.h [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#undef TRACEPOINT_PROVIDER
+#define TRACEPOINT_PROVIDER Nit_Compiler
+
+#undef TRACEPOINT_INCLUDE
+#define TRACEPOINT_INCLUDE "/home/olivier/Bureau/nit/src/nit_compile/traces.h"
```
Comment (privat) [inline clib/traces.h]: According to the doc, a relative path is ok. So `./traces.h`seems enough

File: clib/traces.c [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#define TRACEPOINT_CREATE_PROBES
+#define TRACEPOINT_DEFINE
+
+#include "traces.h"
```
Comment (privat) [inline clib/traces.c]: Remove the C file if it is useless

File: src/compiler/abstract_compiler.nit [hunk 3]
```diff
@@ -75,6 +75,8 @@ redef class ToolContext
 	var opt_release = new OptionBool("Compile in release mode and finalize application", "--release")
 	# -g
 	var opt_debug = new OptionBool("Compile in debug mode (no C-side optimization)", "-g", "--debug")
+	# -t
+	var opt_trace = new OptionBool("Compile with lttng's instrumentation", "-t", "--trace")
```
Comment (privat) [inline src/compiler/abstract_compiler.nit]: Keep only the long option.

**Comment by privat** [comment]
Jenkins : OK to test

**Comment by privat** [comment]
also: http://gresil.org/jenkins/job/CI_github/5730/

* add the option to the manpage https://github.com/nitlang/nit/blob/master/share/man/nitc.md
* do not commit bad whitespaces

File: src/compiler/abstract_compiler.nit [hunk 4]
```diff
@@ -233,6 +236,15 @@ class MakefileToolchain
 		compiler.files_to_copy.add "{clib}/gc_chooser.c"
 		compiler.files_to_copy.add "{clib}/gc_chooser.h"
 
+		# Add lttng traces provider to external bodies
+		if toolcontext.opt_trace.value then
+			var traces = new ExternCFile("traces.c", "-llttng-ust -ldl -I.")
```
[the remark this answers is not in the document]
Comment (olivierschirm) [reply src/compiler/abstract_compiler.nit]: -I. search the directory which was current when gcc was invoked. That solve the issue with the TRACEPOINT_INCLUDE "usr/olivier/bureau.../traces.h". -I. flag make this "./traces..h" work.

[the remark this answers is not in the document]
Comment (privat) [reply src/compiler/abstract_compiler.nit]: OK. keep the -I. and add a comment explaining that

**Pull request merged**